CRM/Civicase/Service/CaseCategoryCustomDataType.php

```diff
@@ -42,7 +42,91 @@ public function create($caseCategoryName) {
   }
 
   /**
-   * Creates the Custom data type option value for case category.
+   * Activates the Custom data type option value for case category.
+   *
+   * Also activates any related custom groups.
+   *
+   * @param string $caseCategoryName
+   *   Case Category Name.
+   */
+  public function activate($caseCategoryName) {
+    $result = $this->getCustomDataOptionValue($caseCategoryName);
+
+    if ($result['count'] == 0) {
+      return;
+    }
+
+    $isActive = TRUE;
+    CRM_Core_BAO_OptionValue::setIsActive($result['values'][0]['id'], $isActive);
+
+    // Activates related custom groups.
+    $result = civicrm_api3('CustomGroup', 'get', [
+      'sequential' => 1,
+      'extends_entity_column_id' => $result['values'][0]['value'],
+      'extends' => 'Case',
+      'options' => ['limit' => 0],
+      'return' => 'id',
+    ]);
+
+    if ($result['count'] == 0) {
+      return;
+    }
+
+    $customGroupIDs = array_column($result['values'], 'id');
+    foreach ($customGroupIDs as $customGroupID) {
+      $customGroup = new CRM_Core_BAO_CustomGroup();
+      $customGroup->id = $customGroupID;
+      // We used is_reserved instead of is_active to show
+      // them in the UI when opening "Custom Data" page.
+      $customGroup->is_reserved = !$isActive;
+      $customGroup->save();
+    }
```

Comment:
This class is responsible for creating/deleting CaseCategoryCustomDataType option values for the `'custom_data_type'` option group, it does not know anything regarding custom groups for the case type category. 
This violates the single responsibility principle. 
The proper way to do this is to have a new service that handles the disabling of custom groups for case categories separately.

That being said, I don't think we need to bother about setting option these values active or inactive, it's fine to leave these data as long as the extensions is still present on the site, either enabled or disabled.  What we can do however is to remove these data once the extension is uninstalled. 

Same comment goes for the deactivate function in this class.
Also, I don't think we need the two functions added here, the `custom_data_type` option group is created by civi itself and it's not really worth disabling these option values since it doesn't affect anything else (unlike the cg_extends) option values in the system, what we can do however is to make sure that these options are removed when the extension is uninstalled. The less things we have to do when enabling/disabling, the better.